workers/go/kitchensink/kitchen_sink.go

```diff
@@ -116,25 +124,61 @@ func handleAction(
 		if child.WorkflowType != "" {
 			childType = child.WorkflowType
 		}
-		err := workflow.ExecuteChildWorkflow(ctx, childType, child.GetInput()).Get(ctx, nil)
+		err := withAwaitableChoice(ctx, func(ctx workflow.Context) workflow.Future {
+			return workflow.ExecuteChildWorkflow(ctx, childType, child.GetInput()[0])
+		}, child.AwaitableChoice)
+		return nil, err
+	} else if patch := action.GetSetPatchMarker(); patch != nil {
+		if workflow.GetVersion(ctx, patch.GetPatchId(), workflow.DefaultVersion, 1) == 1 {
+			return handleAction(ctx, workflowState, patch.GetInnerAction())
+		}
+	} else if setWfState := action.GetSetWorkflowState(); setWfState != nil {
+		workflowState = setWfState
+	} else if awaitState := action.GetAwaitWorkflowState(); awaitState != nil {
+		err := workflow.Await(ctx, func() bool {
+			if val, ok := workflowState.Kvs[awaitState.Key]; ok {
+				return val == awaitState.Value
+			}
+			return false
+		})
+		return nil, err
+	} else if upsertMemo := action.GetUpsertMemo(); upsertMemo != nil {
+		convertedMap := make(map[string]interface{}, len(upsertMemo.GetUpsertedMemo().Fields))
+		for k, v := range upsertMemo.GetUpsertedMemo().Fields {
+			convertedMap[k] = v
+		}
+		err := workflow.UpsertMemo(ctx, convertedMap)
+		return nil, err
+	} else if upsertSA := action.GetUpsertSearchAttributes(); upsertSA != nil {
+		convertedMap := make(map[string]interface{}, len(upsertSA.GetSearchAttributes()))
+		for k, v := range upsertSA.GetSearchAttributes() {
+			convertedMap[k] = v
+		}
+		err := workflow.UpsertSearchAttributes(ctx, convertedMap)
 		return nil, err
 	} else if action.GetNestedActionSet() != nil {
-		return handleActionSet(ctx, action.GetNestedActionSet())
+		return handleActionSet(ctx, workflowState, action.GetNestedActionSet())
 	} else {
 		return nil, fmt.Errorf("unrecognized action")
 	}
 	return nil, nil
 }
 
 func launchActivity(ctx workflow.Context, act *kitchensink.ExecuteActivityAction) error {
+	args := act.GetArguments()
+	if len(args) == 0 {
+		args = nil
+	}
```

Comment:
Is this needed?

Reply:
Not anymore. It was while the pass through converter I added wasn't working right.